test/e2e/multicast_test.go

```diff
@@ -158,6 +170,553 @@ type multicastTestcase struct {
+							},
+						},
+					},
+					Ports: []crdv1alpha1.NetworkPolicyPort{
+						{
+							Protocol: &protocol,
+						},
+					},
+					Action: &dropAction,
+					Name:   "block-multicast-traffic",
+				},
+				{
+					To: []crdv1alpha1.NetworkPolicyPeer{
+						{
+							IPBlock: &crdv1alpha1.IPBlock{
+								CIDR: fmt.Sprintf("%s/32", groupAddressAllow),
+							},
+						},
+					},
+					Ports: []crdv1alpha1.NetworkPolicyPort{
+						{
+							Protocol: &protocol,
+						},
+					},
+					Action: &allowAction,
+					Name:   "allow-multicast-traffic",
+				},
+			},
+		},
+	}
+	if _, err = k8sUtils.CreateOrUpdateANP(anp2); err != nil {
+		failOnError(fmt.Errorf("create ANP failed for ANP %s: %v", anp2.Name, err), t)
+	}
+	// Wait for the policy to be realized before attempting connections
+	failOnError(data.waitForANPRealized(t, data.testNamespace, anp2.Name), t)
+
+	go func() {
+		receiverMulticastAllowCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -o -p %d  -t 3  -W %d %s", 2234, mcjoinWaitTimeout, groupAddressAllow)}
+		data.RunCommandFromPod(data.testNamespace, receiverName1, mcjoinContainerName, receiverMulticastAllowCommand)
+	}()
+	go func() {
+		receiverMulticastDropCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -o -p %d  -t 3  -W %d %s", 2234, mcjoinWaitTimeout, groupAddressDrop)}
+		data.RunCommandFromPod(data.testNamespace, receiverName1, mcjoinContainerName, receiverMulticastDropCommand)
+	}()
+	go func() {
+		receiverMulticastAllowCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -o -p %d  -t 3 -W %d %s", 2234, mcjoinWaitTimeout, groupAddressAllow)}
+		data.RunCommandFromPod(data.testNamespace, receiverName2, mcjoinContainerName, receiverMulticastAllowCommand)
+	}()
+	go func() {
+		receiverMulticastDropCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -o -p %d  -t 3  -W %d %s", 2234, mcjoinWaitTimeout, groupAddressDrop)}
+		data.RunCommandFromPod(data.testNamespace, receiverName2, mcjoinContainerName, receiverMulticastDropCommand)
+	}()
+
+	sendSessions := 10
+	var wg sync.WaitGroup
+	for i := 0; i < sendSessions; i++ {
+		wg.Add(1)
+		go func() {
+			sendMulticastDropCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -f 500 -o -p %d -s -t 3 -c 1 -W %d %s", 2233, mcjoinWaitTimeout, groupAddressDrop)}
+			data.RunCommandFromPod(data.testNamespace, senderName1, mcjoinContainerName, sendMulticastDropCommand)
+			data.RunCommandFromPod(data.testNamespace, senderName2, mcjoinContainerName, sendMulticastDropCommand)
+			sendMulticastAllowCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -f 500 -o -p %d -s -t 3 -c 1 -W %d %s", 2233, mcjoinWaitTimeout, groupAddressAllow)}
+			data.RunCommandFromPod(data.testNamespace, senderName2, mcjoinContainerName, sendMulticastAllowCommand)
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+
+	if err := wait.Poll(5*time.Second, defaultTimeout, func() (bool, error) {
+		stdout, stderr, err := runAntctl(antreaPodReceiver1, antctlCmds, data)
```

Comment:
Use a general function for the logic on runAntctl and check result, and pass different paramaters for the cases?

Reply:
updated